src/components/form/file-upload/file-upload.tsx

```diff
@@ -0,0 +1,184 @@
+
+export const FileUpload: FC<FileUploadProps> = ({
+    label = '',
+    labelFileSize = '',
+    labelButton = '',
+    Icon,
+    IconActionLeft,
+    IconActionRight,
+    onLoadFile,
+    labelCancelButton = '',
+    labelSaveButton = '',
+    maxFileSizeUpload,
+}) => {
+    const fileInputId = useId();
+    const inputReference = useRef<HTMLInputElement>(null);
+
+    const [{ isDragIsOver, isValidFileSize, isValidFileType }, setState] = useState(
+        defaultObserveFileUploadContext(),
+    );
+
+    const handleDragOver = (event: DragEvent<HTMLDivElement>) => {
+        event.preventDefault();
+        setState((prevState) => ({ ...prevState, isDragIsOver: true }));
+    };
+
+    const handleDragLeave = (event: DragEvent<HTMLDivElement>) => {
+        event.preventDefault();
+        setState((prevState) => ({ ...prevState, isDragIsOver: false }));
+    };
+
+    const handleDrop = (event: DragEvent<HTMLDivElement>) => {
+        event.preventDefault();
+        setState((prevState) => ({ ...prevState, isDragIsOver: false }));
+
+        // Fetch the file and check the validation
+        const droppedFile = event.dataTransfer.files[0];
+        droppedFile && checkFileType(droppedFile.type);
+        droppedFile && checkFileSize(droppedFile.size);
+
+        if (droppedFile && isValidFileType && isValidFileSize) {
+            onLoadFile && onLoadFile(droppedFile);
+        }
+
+        // Use FileReader to read file content
+        const reader = new FileReader();
+
+        reader.onloadend = () => {
+            console.log(reader.result);
+        };
+
+        reader.onerror = () => {
+            console.error('There was an issue reading the file.');
+        };
+
+        reader.readAsDataURL(droppedFile);
+        console.log('reader', reader);
+    };
+
+    // Fire the input when the Button is clicked
+    const onFileUpload = () => {
+        inputReference.current?.click();
+    };
+
+    const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+        event.preventDefault();
+
+        event.target.files && checkFileType(event.target.files[0].type);
+        event.target.files && checkFileSize(event.target.files[0].size);
+
+        if (event.target.files && event.target.files[0] && isValidFileType && isValidFileSize) {
+            onLoadFile && onLoadFile(event.target.files[0]);
+        }
+    };
+
+    const checkFileType = (type: string): void => {
+        if (type === 'image/png') {
+            setState((prevState) => ({ ...prevState, isValidFileType: true }));
+        } else if (type === 'image/jpeg') {
+            setState((prevState) => ({ ...prevState, isValidFileType: true }));
+        } else {
+            setState((prevState) => ({ ...prevState, isValidFileType: false }));
+        }
+    };
+
+    const checkFileSize = (size: number): void => {
+        // File Size must be less than 5 MB
+        return size <= maxFileSizeUpload && maxFileSizeUpload
```

Comment:
remove comment or exchange to `// File Size must be less than maxFileSizeUpload `

is `&& maxFileSizeUpload` needed? Since `maxFileSizeUpload` seems to be a mandatory prop